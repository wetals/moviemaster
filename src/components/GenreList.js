import React from "react";

const GenreList = props => {
  const updateGenres = event => {
    props.updateGenres(event.target.value);
  };

  return (
    <div className="genre__list">
      <h3 className="title">All Genres</h3>
      {props.genres &&
        props.genres.map(genre => (
          <div key={genre.id}>
            <label className="checkbox">
              {genre.name}
              <input
                type="checkbox"
                name={genre.name}
                value={genre.id}
                onChange={updateGenres}
              />
              <span className="checkmark" />
            </label>
          </div>
        ))}
    </div>
  );
};

export default GenreList;
