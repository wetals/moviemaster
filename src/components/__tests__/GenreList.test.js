import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import GenreList from "../GenreList";

const testData = {
  genres: [
    {
      id: 1,
      name: "one"
    },
    {
      id: 2,
      name: "two"
    }
  ],
  updateGenres: {
    1: "foo",
    2: "bar"
  }
};

describe("GenreList component", () => {
  it("renders properly", () => {
    const totalGenres = testData.genres.length;
    const wrapper = shallow(<GenreList {...testData} />);
    expect(wrapper.find("label").length).toBe(totalGenres);
    expect(wrapper.find("input").length).toBe(totalGenres);
    expect(wrapper.find("input").get(0).props.name).toBe("one");
  });

  it("renders snapshot properly", () => {
    const tree = renderer.create(<GenreList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
