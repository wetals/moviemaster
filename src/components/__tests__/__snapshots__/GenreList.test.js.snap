// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList component renders snapshot properly 1`] = `
<div
  className="genre__list"
>
  <h3
    className="title"
  >
    All Genres
  </h3>
</div>
`;
